src/tss2-fapi/ifapi_keystore.c

```diff
@@ -75,7 +75,7 @@ initialize_explicit_key_path(
     *list_node1 = split_string(ipath, IFAPI_FILE_DELIM);
     NODE_STR_T *list_node = *list_node1;
     char const *profile;
-    char *hierarchy;
+    char *hierarchy = NULL;
```

Comment:
Maybe we could initialize the hierarchy to the default "HS" here? This could allow us to remove the if (list_node == NULL) check in line 101, especially that the same checked is in line 82, no?

Reply:
I will remove the default setting in line 101. HS should only be set if  SRK is used.
Thank you for the hint.